asserts/snap_asserts.go

```diff
@@ -314,17 +331,113 @@ func assembleSnapDeclaration(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
+	var ras []*RevisionAuthority
+
+	ra, ok := assert.headers["revision-authority"]
+	if ok {
+		ramaps, ok := ra.([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("revision-authority stanza must be a list of maps")
+		}
+		if len(ramaps) == 0 {
+			// there is no syntax producing this scenario but be robust
+			return nil, fmt.Errorf("revision-authority stanza cannot be empty")
+		}
+		ras = make([]*RevisionAuthority, 0, len(ramaps))
+		for _, ramap := range ramaps {
+			m, ok := ramap.(map[string]interface{})
+			if !ok {
+				return nil, fmt.Errorf("revision-authority stanza must be a list of maps")
+			}
+			accountID, err := checkStringMatchesWhat(m, "account-id", "in revision authority", validAccountID)
+			if err != nil {
+				return nil, err
+			}
+			// XXX validity pattern?
+			prov, err := checkStringListInMap(m, "provenance", "provenance in revision authority", nil)
+			if err != nil {
+				return nil, err
+			}
+			if len(prov) == 0 {
+				return nil, fmt.Errorf("provenance in revision authority cannot be empty")
+			}
+			minRevision := 1
+			maxRevision := 0
+			if _, ok := m["min-revision"]; ok {
+				var err error
+				minRevision, err = checkSnapRevisionWhat(m, "min-revision", "in revision authority")
+				if err != nil {
+					return nil, err
+				}
+			}
+			if _, ok := m["max-revision"]; ok {
+				var err error
+				maxRevision, err = checkSnapRevisionWhat(m, "max-revision", "in revision authority")
+				if err != nil {
+					return nil, err
+				}
+			}
+			if maxRevision != 0 && maxRevision < minRevision {
+				return nil, fmt.Errorf("optional max-revision must greater than min-revision in revision-authority")
```

Comment:
Just double checking that the `if` condition is correct, since the message here says "greater" but in the condition you have is a strict `<`.
Maybe, to avoid any confusion, the error could be changed to "optional max-revision can't be less than min-revision in revision-authority".